### ISSUE TITLE
Forced to pass the databaseName

### ISSUE THREAD

**Issue by sylvainar** [seppevs/migrate-mongo#46]
**Describe the bug**
I have multiple env, I don't have to select manually a database, because I put my database name in the connection string, like so:

```
mongodb://localhost:27017/my_db
```

So I tried removing the warning, and it works, thanks to Mongodb client:
```
/**
 …
 * @param {string} [dbName] The name of the database we want to use. If not provided, use database name from connection string.
…
 */
MongoClient.prototype.db = function(dbName, options) {
…
```

So the database name shouldn't be required.

**Expected behavior**
I should be able to use a connection string to select the database, without having the cli popping an error and stopping execution.

**Comment by jsmith-phoenix** [comment]
Adding a +1 on this request as I've not been able to find a way to successfully connect to Mongo Atlas except but using the same modification of passing the full Connection string with DB and empty string for the DatabaseName.

**Comment by sylvainar** [comment]
Temporary workaround was to parse the connection string, twice.

```js
const { ConnectionStringParser } = require('connection-string-parser');

const mongoConnectionString = process.env.DB_URL;

const connectionStringParser = new ConnectionStringParser({
  // To support mongodb atlas connectionstring too
  scheme: mongoConnectionString.startsWith('mongodb+srv://') ? 'mongodb+srv' : 'mongodb',
  hosts: [],
});

// Parse the connection string to get the database name.
const { endpoint } = connectionStringParser.parse(mongoConnectionString);

const config = {
  mongodb: {
    url: mongoConnectionString,

    databaseName: endpoint,

    options: {
      useNewUrlParser: true,
    },
  },

  // The migrations dir, can be an relative or absolute path.
  migrationsDir: 'migrations',

  // The mongodb collection where the applied changes are stored.
  changelogCollectionName: 'changelog',
};

// Return the config as a promise
module.exports = config;

```

**Comment by DanielRuf** [comment]
Much more elegant imo: https://github.com/seppevs/migrate-mongo/pull/115/files

**Comment by sylvainar** [comment]
Sure but the simplest way would be to remove the `throw`, since mongo can parse the database on its own

**Comment by DanielRuf** [comment]
Not in this case as the db is explicitely selected. This might need some refactoring.

The error is there to mark the field as required as described in the commit message of the change.

**Comment by DanielRuf** [comment]
We use some dotenv solution and load the right config based on the environment setting which should be thevrecommended way. And you can always use `process.env` values.

**Comment by sylvainar** [comment]
Yes but do you use a connection string like so: `mongodb://localhost:27017/my_db` ?

**Comment by DanielRuf** [comment]
I do not completely understand what you mean.
My PR supports this as it parses it, fallback is the default `admin` database.

**Comment by buccfer** [comment]
@sylvainar I think a more simpler workaround is using the native `url` module of Node:

```
const url = require('url');
const { DB_URL } = process.env;

const {pathname} = url.parse(DB_URL);
const databaseName = pathname.substring(1);

module.exports = {
  mongodb: {
    url: DB_URL,
    databaseName,
    options: {
      useNewUrlParser: true,
      useUnifiedTopology: true,
      connectTimeoutMS: 36e5, // 1 hour.
      socketTimeoutMS: 36e5 // 1 hour.
    }
  },
  migrationsDir: 'migrations',
  changelogCollectionName: 'changelog'
};
```

**Comment by seppevs** [comment]
Fixed in v8.1.1
I'm closing this issue. Please reopen when necessary.
